workflows/README.md

```diff
@@ -0,0 +1,11 @@
+# Workflows
+
+A workflow is custom application logic that consists of a set of tasks and or state transitions.
```

Comment:
Should we add a link to the proposal so that readers can get more context on this feature area?